tests/testthat/test-verb-select.R

```diff
@@ -35,6 +35,31 @@ test_that("select after distinct produces subquery", {
   expect_true(lq$x$distinct)
 })
 
+test_that("select after arrange produces subquery, if needed", {
+  lf <- lazy_frame(x = 1)
+
+  # shouldn't inline
+  out <- lf %>% mutate(z = 2) %>% arrange(x, z) %>% select(x)
```

Comment:
I'm not sure this is testing the right thing because when I run it, I see:

```
SELECT `x`
FROM (
  SELECT `df`.*, 2.0 AS `z`
  FROM `df`
) AS `q01`
Warning message:
ORDER BY is ignored in subqueries without LIMIT
ℹ Do you need to move arrange() later in the pipeline or use window_order() instead? 
```

Reply:
Or maybe that is the expected behaviour? But definitely worth a snapshot test, I think.

Reply:
Yeah that is expected. I've re-added the snapshot I removed in my prior commit.